compact_test.go

```diff
@@ -1058,3 +1060,38 @@ func TestDeleteCompactionBlockAfterFailedReload(t *testing.T) {
 		})
 	}
 }
+
+// TestOpenBlockWithHook ensures that when OpenBlock call rename failed, there is not meta.json file
+// It use fuse to inject rename filesystem error
+func TestOpenBlockWithHook(t *testing.T) {
+	// Init action.
+	original, err := ioutil.TempDir("", "dev")
+	testutil.Ok(t, err)
+	mountpoint, err := ioutil.TempDir("", "mountpoint")
+	testutil.Ok(t, err)
+
+	// Create block will be successful because hook server does not start.
+	_, file := filepath.Split(createBlock(t, original, genSeries(1, 1, 200, 300)))
+	clean := fuse.NewServer(t, original, mountpoint, fuse.Hook(fuse.TestRenameHook{}))
```

Comment:
instead of `clean` you can have `func (h *hookFs) Close() error { return server.Unmount() }` so it becomes like 

```
server,error := fuse.NewServer(original, mountpoint, fuse.FailingRenameHook{})
testutil.Ok(t, err)
defer func(){
	testutil.Ok(t, server.Close())
}()
```

Note that I have taken the t testing.T out of the NewServer arguments. best to keep this in the tests only.

Note that you don't need casting for `fuse.Hook(fuse.FailingRenameHook{}) -> fuse.FailingRenameHook{}`

Reply:
Accept and I will updated it.

@krasi-georgiev , is it convenient to add a contact information?

Reply:
Hi @krasi-georgiev 
If we take the t testing.T out of the NewServer arguments, the test cases will like this:
```
	server, err := fuse.NewServer(original, mountpoint, fuse.FailingRenameHook{})
	// Remember to call unmount after you do not use it.
	defer func() {
		if server != nil {
			testutil.Ok(t, server.Close())
		}
	}()

	if err != nil {
		t.Skip()
	}
```
These codes will be redundant, i think it is better to do this check only in NewServer method, not in every test cases.

Reply:
> is it ok to add a contact information?

not sure what you mean by this.


> These codes will be redundant, i think it is better to do this check only in NewServer method, not in every test cases.

It is fine to do this check with every test. Even with a bit of repetitive code is better for being more explicit and show why the test is skipped instead if hiding it in `fuse.NewServer`



this is how I see the final result.

```
server, err := fuse.NewServer(original, mountpoint, fuse.FailingRenameHook{})
if err != nil {
	t.Skip("use server couldn't be started")
}	
defer func() {
	testutil.Ok(t, server.Close())
}()

```